terraform/modules/monitoring/monitoring.tf

```diff
@@ -0,0 +1,135 @@
+variable "gcp_project" {
+  type = string
+}
+
+variable "gcp_region" {
+  type = string
+}
+
+resource "kubernetes_namespace" "monitor" {
+  metadata {
+    name = "monitor"
+  }
+}
+
+resource "kubernetes_service" "prometheus" {
+  metadata {
+    name      = "prometheus"
+    namespace = "monitor"
+  }
+  spec {
+    selector = {
+      app = kubernetes_pod.prometheus.metadata.0.labels.app
+    }
+    port {
+      port        = 9090
+      target_port = 9090
+    }
+
+    type = "LoadBalancer"
+  }
+}
+
+resource "kubernetes_pod" "prometheus" {
+  metadata {
+    name      = "prometheus"
+    namespace = "monitor"
+    labels = {
+      app = "Prometheus"
+    }
+  }
+
+  spec {
+    service_account_name = kubernetes_service_account.prometheus_account.metadata[0].name
+    container {
+      image = "prom/prometheus:latest"
+      name  = "prometheus"
+    }
+    volume {
+      name = "config-volume"
+      config_map {
+        name = "prometheus_config"
+      }
+    }
+  }
+}
+
+resource "kubernetes_config_map" "prometheus_config" {
+  metadata {
+    name = "prometheus-config"
+  }
+
+  data = {
+    "prometheus.yml" = file("${path.module}/prometheus.yml")
```

Comment:
nit: it's a shame having some of this farmed out to a separate YAML file, especially since some of the values in there need to agree with values here in Terraform (e.g., the port numbers). I think you could use [yamlencode](https://www.terraform.io/docs/configuration/functions/yamlencode.html) to instead define all those parameters inline in Terraform.

Reply:
FWIW, I am more used to maintaining Prometheus config as its own YAML, and prefer it that way. For instance if we need to copy a config block from some example, we don't need to transmute it into HCL first.

The one parameter that needs agreement is the port, and 9090 is a standard port for Prometheus, so I'm not worried about writing it twice.